### ISSUE TITLE
Building and Installing Occlum-HyperEnclave

### ISSUE THREAD

**Issue by vramasw3** [HyperEnclave/hyperenclave#32]
Hey there, I was exploring different TEEs and came across the HyperEnclave paper. I found it very interesting and wanted to experiment with it. 

I see in the docker container that Occlum is already built and installed. When attempting to make Occlum without any changes, I find that I am unable to get Occlum to build the make submodule to prepare a make on Occlum itself, I get this error:

```
error[E0658]: use of unstable library feature 'mixed_integer_ops'
   --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/unicode-width-0.1.14/src/tables.rs:407:22
    |
407 |                 (sum.wrapping_add_signed(isize::from(add)), info)
    |                      ^^^^^^^^^^^^^^^^^^^
    |
    = note: see issue #87840 <https://github.com/rust-lang/rust/issues/87840> for more information
    = help: add `#![feature(mixed_integer_ops)]` to the crate attributes to enable

error[E0658]: use of unstable library feature 'mixed_integer_ops'
   --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/unicode-width-0.1.14/src/tables.rs:702:22
    |
702 |                 (sum.wrapping_add_signed(isize::from(add)), info)
    |                      ^^^^^^^^^^^^^^^^^^^
    |
    = note: see issue #87840 <https://github.com/rust-lang/rust/issues/87840> for more information
    = help: add `#![feature(mixed_integer_ops)]` to the crate attributes to enable

   Compiling sgx_types v1.1.4 (/root/occlum/deps/rust-sgx-sdk/sgx_types)
For more information about this error, try `rustc --explain E0658`.
   Compiling termcolor v1.4.1
error: could not compile `unicode-width` due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
error: build failed
```
When running the build command "OCCLUM_RELEASE_BUILD = 1 make", as per the Occlum compilation instructions in their documentation, I have no problems making Occlum.

However, when using "make install", I get this error:

```
CC <= /root/occlum/build/internal/src/pal/src_sim/base64.o
CC <= /root/occlum/build/internal/src/pal/src_sim/errno2str.o
src/ocalls/attestation.c:2:10: fatal error: sgx_dcap_ql_wrapper.h: No such file or directory
    2 | #include <sgx_dcap_ql_wrapper.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [Makefile:83: /root/occlum/build/internal/src/pal/src_sim/ocalls/attestation.o] Error 1
make[1]: *** [Makefile:5: all] Error 2
make: *** [Makefile:118: install_bins_and_libs] Error 2
```

TLDR:
Trying to make and install an unmodified Occlum-HE from the source within the container, finding build errors.

Thank you for your time!

**Comment by Bonjourz** [comment]
Hi @Vikrius ,

We need some time to analysis the issue.

By the way, this repository has been moved to a new location: https://github.com/asterinas/hyperenclave, and will be will be archived later.

Could you please copy it there: https://github.com/asterinas/hyperenclave/issues, and have a discussion?

Thx.
